### ISSUE TITLE
SwiftFormat crashes with `illegal hardware instruction`

### ISSUE THREAD

**Issue by tadeuzagallo** [nicklockwood/SwiftFormat#78]
Not sure what information you need, happy to help debug given some pointers. I saved a coredump in any case. Here's the stack trace:

```
* thread #1: tid = 0x0000, 0x000000010b192f81 swiftformat`function signature specialization <Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed, Arg[3] = Owned To Guaranteed> of static swiftformatTool.FormatRules.(wrapArguments (swiftformatTool.Formatter) -> ()).((wrapArguments #1) (for : Swift.Array<Swift.String>, mode : swiftformatTool.WrapMode, allowGrouping : Swift.Bool) -> ()).(closure #2) + 6273, stop reason = signal SIGSTOP
  * frame #0: 0x000000010b192f81 swiftformat`function signature specialization <Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed, Arg[3] = Owned ToGuaranteed> of static swiftformatTool.FormatRules.(wrapArguments (swiftformatTool.Formatter) -> ()).((wrapArguments #1) (for : Swift.Array<Swift.String>, mode : swiftformatTool.WrapMode, allowGrouping : Swift.Bool) -> ()).(closure #2) + 6273
    frame #1: 0x000000010b166441 swiftformat`static swiftformatTool.FormatRules.(wrapArguments (swiftformatTool.Formatter) -> ()).((wrapArguments #1) (for : Swift.Array<Swift.String>, mode : swiftformatTool.WrapMode, allowGrouping : Swift.Bool) -> ()).(closure #1) with unmangled suffix "_merged" + 49
    frame #2: 0x000000010b19595c swiftformat`partial apply forwarder for static swiftformatTool.FormatRules.(wrapArguments (swiftformatTool.Formatter) -> ()).((wrapArguments #1) (for : Swift.Array<Swift.String>, mode : swiftformatTool.WrapMode, allowGrouping : Swift.Bool) -> ()).(closure #2) with unmangled suffix "_merged" + 92
    frame #3: 0x000000010b1b1f1a swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed> of swiftformatTool.Formatter.forEachToken (where : (swiftformatTool.Token) -> Swift.Bool, (Swift.Int, swiftformatTool.Token) -> ()) -> () + 490
    frame #4: 0x000000010b18fdff swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[3] = Owned To Guaranteed> of static swiftformatTool.FormatRules.(wrapArguments (swiftformatTool.Formatter) -> ()).(wrapArguments #1) (for : Swift.Array<Swift.String>, mode : swiftformatTool.WrapMode, allowGrouping : Swift.Bool) -> () + 287
    frame #5: 0x000000010b18fed8 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Dead> of static swiftformatTool.FormatRules.wrapArguments (swiftformatTool.Formatter) -> () + 184
    frame #6: 0x000000010b1664d9 swiftformat`@objc static swiftformatTool.FormatRules.wrapArguments (swiftformatTool.Formatter) -> () + 25
    frame #7: 0x000000010b195a14 swiftformat`partial apply forwarder for static swiftformatTool.FormatRules.(byName : Swift.Dictionary<Swift.String, (swiftformatTool.Formatter) -> ()>).(variable initialization expression).(closure #1).(closure #1) + 84
    frame #8: 0x000000010b195a88 swiftformat`partial apply forwarder for reabstraction thunk helper from @callee_owned (@owned swiftformatTool.Formatter) -> () to @callee_owned (@in swiftformatTool.Formatter) -> (@out ()) + 56
    frame #9: 0x000000010b1c75f1 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned ToGuaranteed> of swiftformatTool.format (Swift.Array<swiftformatTool.Token>, rules : Swift.Array<(swiftformatTool.Formatter) -> ()>, options : swiftformatTool.FormatOptions) throws -> Swift.Array<swiftformatTool.Token> + 993
    frame #10: 0x000000010b1c9f85 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[3] = Owned To Guaranteed, Arg[5] = Owned To Guaranteed and Exploded, Arg[6] = Owned To Guaranteed, Arg[7] = Owned To Guaranteed> of swiftformatTool.(processInput (Swift.Array<Foundation.URL>, andWriteToOutput : Swift.Optional<Foundation.URL>, withRules : Swift.Array<(swiftformatTool.Formatter) -> ()>, formatOptions : swiftformatTool.FormatOptions, fileOptions : swiftformatTool.FileOptions, cacheURL : Swift.Optional<Foundation.URL>) throws -> (Swift.Int, Swift.Int)).(closure #1) + 2837
    frame #11: 0x000000010b1bb0cb swiftformat`swiftformatTool.(processInput (Swift.Array<Foundation.URL>, andWriteToOutput : Swift.Optional<Foundation.URL>, withRules : Swift.Array<(swiftformatTool.Formatter) -> ()>, formatOptions : swiftformatTool.FormatOptions, fileOptions : swiftformatTool.FileOptions, cacheURL: Swift.Optional<Foundation.URL>) throws -> (Swift.Int, Swift.Int)).(closure #1) + 459
    frame #12: 0x000000010b1c931e swiftformat`partial apply forwarder for swiftformatTool.(processInput (Swift.Array<Foundation.URL>, andWriteToOutput : Swift.Optional<Foundation.URL>, withRules : Swift.Array<(swiftformatTool.Formatter) -> ()>, formatOptions : swiftformatTool.FormatOptions, fileOptions : swiftformatTool.FileOptions, cacheURL : Swift.Optional<Foundation.URL>) throws -> (Swift.Int, Swift.Int)).(closure #1) + 590
    frame #13: 0x000000010b1c1d40 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed and Exploded, Arg[3] = Owned To Guaranteed> of swiftformatTool.enumerateSwiftFiles (withInputURL : Foundation.URL, outputURL : Swift.Optional<Foundation.URL>, options : swiftformatTool.FileOptions, block : (Foundation.URL, Foundation.URL) throws -> ()) throws -> () + 3600
    frame #14: 0x000000010b1c1933 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed and Exploded, Arg[3] = Owned To Guaranteed> of swiftformatTool.enumerateSwiftFiles (withInputURL : Foundation.URL, outputURL : Swift.Optional<Foundation.URL>, options : swiftformatTool.FileOptions, block : (Foundation.URL, Foundation.URL) throws -> ()) throws -> () + 2563
    frame #15: 0x000000010b1c1933 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed and Exploded, Arg[3] = Owned To Guaranteed> of swiftformatTool.enumerateSwiftFiles (withInputURL : Foundation.URL, outputURL : Swift.Optional<Foundation.URL>, options : swiftformatTool.FileOptions, block : (Foundation.URL, Foundation.URL) throws -> ()) throws -> () + 2563
    frame #16: 0x000000010b1c1933 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed and Exploded, Arg[3] = Owned To Guaranteed> of swiftformatTool.enumerateSwiftFiles (withInputURL : Foundation.URL, outputURL : Swift.Optional<Foundation.URL>, options : swiftformatTool.FileOptions, block : (Foundation.URL, Foundation.URL) throws -> ()) throws -> () + 2563
    frame #17: 0x000000010b1c1933 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed and Exploded, Arg[3] = Owned To Guaranteed> of swiftformatTool.enumerateSwiftFiles (withInputURL : Foundation.URL, outputURL : Swift.Optional<Foundation.URL>, options : swiftformatTool.FileOptions, block : (Foundation.URL, Foundation.URL) throws -> ()) throws -> () + 2563
    frame #18: 0x000000010b1c1933 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed and Exploded, Arg[3] = Owned To Guaranteed> of swiftformatTool.enumerateSwiftFiles (withInputURL : Foundation.URL, outputURL : Swift.Optional<Foundation.URL>, options : swiftformatTool.FileOptions, block : (Foundation.URL, Foundation.URL) throws -> ()) throws -> () + 2563
    frame #19: 0x000000010b1c1933 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed and Exploded, Arg[3] = Owned To Guaranteed> of swiftformatTool.enumerateSwiftFiles (withInputURL : Foundation.URL, outputURL : Swift.Optional<Foundation.URL>, options : swiftformatTool.FileOptions, block : (Foundation.URL, Foundation.URL) throws -> ()) throws -> () + 2563
    frame #20: 0x000000010b1c5fc4 swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed, Arg[3] = Owned To Guaranteed, Arg[4] = Exploded> of swiftformatTool.processInput (Swift.Array<Foundation.URL>, andWriteToOutput : Swift.Optional<Foundation.URL>, withRules : Swift.Array<(swiftformatTool.Formatter) -> ()>, formatOptions : swiftformatTool.FormatOptions, fileOptions : swiftformatTool.FileOptions, cacheURL : Swift.Optional<Foundation.URL>) throws -> (Swift.Int, Swift.Int) + 4564
    frame #21: 0x000000010b15a1ef swiftformat`function signature specialization <Arg[0] = Owned To Guaranteed> of swiftformatTool.processArguments (Swift.Array<Swift.String>) -> () + 8879
    frame #22: 0x000000010b150fb6 swiftformat`main + 118
    frame #23: 0x00007fff9c8885ad libdyld.dylib`start + 1
```

EDIT: the command I was running was:

```
swiftformat --indent 4 --allman false --wraparguments afterfirst --removelines disabled --wrapelements beforefirst --header ignore --insertlines disabled --empty tuples --ranges nospace --trimwhitespace always --linebreaks lf --commas inline --comments ignore --ifdef indent --hexliterals lowercase --semicolons inline .
```

**Comment by tadeuzagallo** [comment]
We chatted offline, but I'll leave it here for the sake of history in case someone finds it later:

It seems that the issue happens when you have a comment immediately before a closing brace.

Minimal repro:

```swift
a(//comment
)
```

**Comment by nicklockwood** [comment]
Fixed in 0.23. Thanks!
